app/controllers/sessions.rb

```diff
@@ -0,0 +1,21 @@
+get '/sessions/new' do
+  erb :"sessions/new"
+end
+
+post '/sessions' do
+  email, password = params[:email], params[:password]
+  user = User.authenticate(email, password)
+  if user
+    session[:user_id] = user.id
+    redirect to('/')
+  else
+    flash[:errors] = ["The email or password is incorrect"]
```

Comment:
The way it is now, flash[:errors] persists for two requests. So, if I try to login with wrong credentials an error will be displayed and when I use the right credentials the second time I will logon successfully but I will see the error from before. To fix this, we can write flash.now[:errors] = ... and now, the flash message will persist only for one request.